hal/inc/ble_hal.h

```diff
@@ -312,6 +319,14 @@ typedef struct hal_ble_pairing_status_updated_evt_t {
     uint8_t reserved[3];
 } hal_ble_pairing_status_updated_evt_t;
 
+typedef struct hal_ble_pairing_auth_data_t {
+    hal_ble_pairing_auth_data_type_t type;
+    union {
+        bool equal; //For numeric comparison result
+        const uint8_t* data; // For passkey and OOB data
```

Comment:
Might be a good idea to wrap this into a struct or a pointer to a struct?

Reply:
For passkey and legacy OOB data, they are plain data with fixed length, for LESC OOB data, we can static cast the `uint8*` to corresponding struct pointer as well. Maybe extend the union struct like this?:
```
typedef struct hal_ble_pairing_auth_data_t {
    hal_ble_pairing_auth_data_type_t type;
    uint16_t size;
    union {
        bool equal; //For numeric comparison result
        const char* passkey; // For passkey
        const uint8_t* legacy_oob; // For legacy OOB data
        const uint8_t* lesc_oob; // For LESC OOB data
    } params;
} hal_ble_pairing_auth_data_t;
```

Reply:
:+1: